### PR TITLE
Added basic survey editing.

### PR DESCRIPTION
**Pull request by Hyperkid123**
### Changes
- added survey editor to portfolio items
- editing works

### What is missing
- ~~turn off adding and changing order of fields (will be done in DDF)~~ done
- turn of removing fields
- validation editing (Will be done in DFF)
- visual design (Susan is working on that)

survey editing

### CODE REVIEW

**Comment by Hyperkid123** [comment]
@gmcculloug since the Visuals are obviously not complete and it will take some time to do design them, should we hide the link and leave the route there? I would do that for now.

**Comment by gmcculloug** [comment]
Good start, excited to see where this goes.

File: src/smart-components/portfolio/portfolio-item-detail/detail-toolbar-actions.js [hunk 1]
```diff
@@ -93,6 +94,15 @@ const DetailToolbarActions = ({
                   </Link>
                 }
                 role="link"
+              />,
+              <DropdownItem
+                aria-label="Rdit survey"
```
Comment (lgalis) [inline src/smart-components/portfolio/portfolio-item-detail/detail-toolbar-actions.js]: Typo - 'Rdit'

File: src/helpers/shared/user-login.js [hunk 2]
```diff
@@ -138,6 +144,10 @@ export function getIconApi() {
   return iconApi;
 }
 
+export function getServicePlainsApi() {
```
Comment (lgalis) [inline src/helpers/shared/user-login.js]: typo - getServicePlansApi, not getServicePlainsApi

File: src/smart-components/survey-editing/survey-editor.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+import FormBuilder from '@data-driven-forms/form-builder';
+import {
+  builderMapper,
+  fieldProperties,
+  pickerMapper,
+  propertiesMapper
+} from '@data-driven-forms/form-builder/dist/pf4-builder-mappers';
+import {
+  getAxiosInstance,
+  getServicePlainsApi
```
Comment (lgalis) [inline src/smart-components/survey-editing/survey-editor.js]: type, as above - getServicePlansApi

File: src/smart-components/survey-editing/survey-editor.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  const [editedTemplate, setEditedTemplate] = useState({ fields: [] });
+  useEffect(() => {
+    getAxiosInstance()
+      .get(
+        `${CATALOG_API_BASE}/portfolio_items/${portfolioItemId}/service_plans`
+      )
+      .then(([{ create_json_schema: { schema } }]) => setSchema(schema));
+  }, []);
+  const handleSaveSurvey = () => {
+    return getServicePlainsApi()
```
Comment (lgalis) [inline src/smart-components/survey-editing/survey-editor.js]: - same - getServicePlansApi

**Comment by Hyperkid123** [comment]
@lgalis yep this will all happen later. I waiting with that for Susan. We dont event know if we will use normal page of some overlay etc.

Multi select is not supported yet

> I do not see the component chooser at all when opening the editor.

That is intentional. You are not supposed to be able to add/change/remove fields. Removing is still visible though and will change in the future.

**Comment by lgalis** [comment]
@Hyperkid123  - sounds good - I mentioned the component chooser because it is visible in your gif - so that is different than what I saw when testing.

**Comment by gmcculloug** [comment]
cc @sbuenafe-rh

**Comment by Hyperkid123** [comment]
@lgalis i got fix fir the multi select just updating to new package

**Comment by Hyperkid123** [comment]
OK the multi select should be crashing anymore

**Pull request merged**